src/pages/signin/ValidateCodeForm.js

```diff
@@ -0,0 +1,230 @@
+import React from 'react';
+import {
+    TouchableOpacity, View,
+} from 'react-native';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import _ from 'underscore';
+import styles from '../../styles/styles';
+import Button from '../../components/Button';
+import Text from '../../components/Text';
+import themeColors from '../../styles/themes/default';
+import * as Session from '../../libs/actions/Session';
+import ONYXKEYS from '../../ONYXKEYS';
+import CONST from '../../CONST';
+import ChangeExpensifyLoginLink from './ChangeExpensifyLoginLink';
+import withLocalize, {withLocalizePropTypes} from '../../components/withLocalize';
+import compose from '../../libs/compose';
+import TextInput from '../../components/TextInput';
+import * as ValidationUtils from '../../libs/ValidationUtils';
+import withToggleVisibilityView, {toggleVisibilityViewPropTypes} from '../../components/withToggleVisibilityView';
+import canFocusInputOnScreenFocus from '../../libs/canFocusInputOnScreenFocus';
+import * as ErrorUtils from '../../libs/ErrorUtils';
+import {withNetwork} from '../../components/OnyxProvider';
+import networkPropTypes from '../../components/networkPropTypes';
+import OfflineIndicator from '../../components/OfflineIndicator';
+import * as User from '../../libs/actions/User';
+
+const propTypes = {
+    /* Onyx Props */
+
+    /** The details about the account that the user is signing in with */
+    account: PropTypes.shape({
+        /** Whether or not two factor authentication is required */
+        requiresTwoFactorAuth: PropTypes.bool,
+
+        /** Whether or not a sign on form is loading (being submitted) */
+        isLoading: PropTypes.bool,
+    }),
+
+    /** The credentials of the person signing in */
+    credentials: PropTypes.shape({
+        /** The login of the person signing in */
+        login: PropTypes.string,
+    }),
+
+    /** Information about the network */
+    network: networkPropTypes.isRequired,
+
+    ...withLocalizePropTypes,
+    ...toggleVisibilityViewPropTypes,
+};
+
+const defaultProps = {
+    account: {},
+    credentials: {},
+};
+
+class ValidateCodeForm extends React.Component {
+    constructor(props) {
+        super(props);
+        this.validateAndSubmitForm = this.validateAndSubmitForm.bind(this);
+        this.resendValidateCode = this.resendValidateCode.bind(this);
+        this.clearSignInData = this.clearSignInData.bind(this);
+
+        this.state = {
+            formError: false,
+            validateCode: '',
+            twoFactorAuthCode: '',
+        };
+    }
+
+    componentDidMount() {
+        if (!canFocusInputOnScreenFocus() || !this.inputValidateCode || !this.props.isVisible) {
+            return;
+        }
+        this.inputValidateCode.focus();
+    }
+
+    componentDidUpdate(prevProps, prevState) {
+        if (!prevProps.isVisible && this.props.isVisible) {
+            this.inputValidateCode.focus();
+        }
```

Comment:
@Beamanator in `componentDidUpdate` we are not checking for `canFocusInputOnScreenFocus` should not we do so?

```js
    componentDidUpdate(prevProps) {
        if (canFocusInputOnScreenFocus() && !prevProps.isVisible && this.props.isVisible) {
            this.inputValidateCode.focus();
        }
        ...
    }
```